uaclient/config.py

```diff
@@ -288,7 +288,9 @@ def _attached_status(self) -> "Dict[str, Any]":
                 contractInfo["effectiveTo"], "%Y-%m-%dT%H:%M:%SZ"
             )
 
-        resources = get_available_resources(self)
+        resources = self.machine_token["availableResources"]
+        if not resources:
```

Comment:
When would we take this code path?

Reply:
only on current production which hasn't updated yet

Reply:
OK, that makes sense. I think those cases will hit a `KeyError` on the above line as things stand?

Reply:
ohhh right, those will hit a key error as things stand good catch.

Reply:
Fixed.